### PR TITLE
Openapi generator doc 3.x

### LINKED ISSUE #4268: OpenAPITool Helidon Documentation
- [ ] Server MP
- [ ] Server SE
- [ ] Client MP
- [ ] Client SE

See:
#4021
#4020
#4018
#4023

### PR DESCRIPTION
**Pull request by tjquinno**
Resolves #4268 

This PR contains the new material for using the OpenAPI generators for Helidon. This material started with Andrii's PR #4268 and built on it but targets 3.x instead of 4.x.

These new pages are rather long, but there is not much useful explanation on the OpenAPITools website. If we want our users to be productive using the generator and the generated code, I think it’s useful to provide all this information. Still, if you have good suggestions for streamlining the text I'll be very glad to hear them.

Reviewers: I welcome all feedback from everyone, but especially to the OpenAPI tools "sub-team" please look at the parts of the doc that address the work you have done.

### CODE REVIEW

**Review by tvallin — APPROVED** [review]

File: docs/includes/openapi/openapi-generator.adoc [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+As a result, you can use the generator to create code that fits smoothly into your Helidon applications.
+The OpenAPI generator gained particularly strong support for Helidon in release
+{first-version-with-strong-helidon-support}.
+This document applies to that release and later ones.
+
+In the vocabulary of the tool, there are two _generators_ for Helidon:
+
+* `java-helidon-client` (which this document refers to as the Helidon client generator)
+* `java-helidon-server` (Helidon server generator).
+
```
Comment (spericas) [inline docs/includes/openapi/openapi-generator.adoc:51]: Use same structure in parenthesis for parallelism? For example, "hereinafter Helidon ... generator".

File: docs/includes/openapi/openapi-generator.adoc [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+endif::mp-flavor[]
+ifdef::se-flavor[]
+Helidon SE client based on xref:{helidon-client-xref}[Helidon WebClients].
+endif::se-flavor[]
+The resulting client library works with any server that implements the API declared in the OpenAPI document you specified when you ran the generator.
+The client library provides an abstraction similar to remote procedure calls (RPC).
+To access a remote service that implements the endpoints declared in the OpenAPI document, your code uses the generated client library first to establish a connection to the remote service and then to call remote service endpoints by invoking local methods passing POJO business objects or Java types as arguments.
+
+Use the tool's Helidon _server_ generator and its `{flavor-lc}` library to create server endpoint stubs for a Helidon {flavor-uc} service. You build on these stubs by extending a generated class or implementing a generated interface, adding your specific business logic to finish the implementation of the endpoints. The combination of the generated server code plus Helidon {flavor-uc} underneath it allows you to focus on the business details instead of the networking.
```
Comment (spericas) [inline docs/includes/openapi/openapi-generator.adoc:68]: "networking" feels a bit too low level here. Perhaps "resource boilerplate" or "resource method declarations"?

File: docs/includes/openapi/openapi-generator.adoc [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+
+You can run the OpenAPI generators in three ways:
+
+// tag::three-ways-to-run[]
+* using the OpenAPI generator CLI
+* using the OpenAPI generator Maven or Gradle plug-in
```
Comment (spericas) [inline docs/includes/openapi/openapi-generator.adoc:74]: Do we want to mention Gradle here?

File: docs/includes/openapi/openapi-generator.adoc [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+In some cases, you might need more control over the response sent to the client. In that case, specify the additional property `returnResponse=true` when you run the Helidon server generator. The return type for the generated methods is
+ifdef::mp-flavor[]
+the Jakarta RESTful web services `Response`
+endif::mp-flavor[]
+ifdef::se-flavor[]
+the Helidon SE `ServerResponse`
+endif::se-flavor[]
+and your code has complete control--and therefore responsibility--over setting the status, writing the response entity (if any), and assigning any returned headers.
+
+Your code plus the server code from the Helidon generator--all running on Helidon {flavor-uc}--combine to implement fully the server API declared in the original OpenAPI document. Build your project to get a tailored Helidon {flavor-uc} server `.jar` file or Docker image and your server is ready to run.
```
Comment (spericas) [inline docs/includes/openapi/openapi-generator.adoc:525]: "fully implement"?

File: docs/includes/openapi/openapi-generator.adoc [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+- <<Usage, Usage>>
+- <<References, References>>
+
+== Overview
+The link:{openapi-spec-url}[OpenAPI specification] provides a standard way to express RESTful APIs.
+
+Separately, the link:{openapi-generator-tool-site-url}[OpenAPI generator] project has created a powerful code generator tool which accepts an OpenAPI document and generates client and server code for many languages and frameworks. The Helidon team contributes to this tool to ensure that it provides strong support for Helidon {flavor-uc} clients and servers.
+As a result, you can use the generator to create code that fits smoothly into your Helidon applications.
+The OpenAPI generator gained particularly strong support for Helidon in release
+{first-version-with-strong-helidon-support}.
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply docs/includes/openapi/openapi-generator.adoc:44]: Good point. Fixed.

File: docs/includes/openapi/openapi-generator.adoc (hunk 1 above)
Comment (tjquinno) [reply docs/includes/openapi/openapi-generator.adoc:51]: Fixed

File: docs/includes/openapi/openapi-generator.adoc (hunk 2 above)
Comment (tjquinno) [reply docs/includes/openapi/openapi-generator.adoc:68]: Fixed

File: docs/includes/openapi/openapi-generator.adoc (hunk 3 above)
Comment (tjquinno) [reply docs/includes/openapi/openapi-generator.adoc:74]: Fixed

File: docs/includes/openapi/openapi-generator.adoc (hunk 4 above)
Comment (tjquinno) [reply docs/includes/openapi/openapi-generator.adoc:525]: Ah, yes whether to boldly split that infinitive! IIRC Lisa pointed me to the conventions or advised me separately that splitting infinitives is OK in Oracle doc and (to my ear) it does read a bit better with your change. Fixed.

**Review by aserkes — APPROVED** [review]

**Review by spericas — APPROVED** [review]

**Review by tvallin — APPROVED** [review]

**Pull request merged**